MicroBenchmarks/LoopVectorization/ConditionalScalarAssignment.cpp

```diff
@@ -0,0 +1,118 @@
+#include <iostream>
+#include <memory>
+#include <random>
+
+#include "benchmark/benchmark.h"
+
+#define ITERATIONS 10000000
+
+// Find the last element in A above the given threshold,
+// with default loop vectorization settings.
+template <typename T> static T run_csa_autovec(T *A, T *B, T *C, T Threshold) {
+  // Pick a default value that's out of the uniform distribution created
+  // for 'A' in init_data below.
+  T Result = 101;
+  for (unsigned i = 0; i < ITERATIONS; i++) {
+    // Do some work to make the difference noticeable
```

Comment:
could you add a few more variations, like the minimal case with just a CAS and multiple independent CAS?